packages/code-studio/src/main/AppMainContainer.tsx

```diff
@@ -335,6 +340,35 @@ export class AppMainContainer extends Component<
           }
         });
 
+        updated.forEach(toUpdate => {
+          let panelId: string | undefined = undefined;
+
+          if (layout) {
+            const widgetContentItem = LayoutUtils.getContentItemInRoot(
+              layout.root,
+              {
+                title: toUpdate.title,
+              }
+            );
```

Comment:
We shouldn't be using the title here. The title changes when we [clone panels](https://github.com/deephaven/web-client-ui/blob/309ff795dcb367fc9b7b6f2abb4a07b10cf8ab55/packages/dashboard/src/layout/LayoutUtils.ts#L675) and when we [rename](https://github.com/deephaven/web-client-ui/blob/309ff795dcb367fc9b7b6f2abb4a07b10cf8ab55/packages/dashboard/src/layout/LayoutUtils.ts#L647) [panels](https://github.com/deephaven/web-client-ui/blob/309ff795dcb367fc9b7b6f2abb4a07b10cf8ab55/packages/golden-layout/src/items/AbstractContentItem.ts#L345).

We should instead be matching by `panelId` (which we should already know if the panel was opened by `AppMainContainer`) or by the `metadata`. Problem is as you've identified below, `ConsolePanel` is doing a similar thing for panels that it has opened, so you can get conflicts.

Perhaps the panel/component itself should be listening for field updates, and update the widget if the underlying field changes (based somewhat on Raffi's comment that he would always want/expect the widget to reflect the currently named widget, indicating our behaviour when opening a panel and it not updating is incorrect).

Reply:
I like the idea of the panel listening for changes and updating itself. I'll look into that.